src/Symfony/Bridge/PhpUnit/DeprecationErrorHandler/Configuration.php

```diff
@@ -0,0 +1,208 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\PhpUnit\DeprecationErrorHandler;
+
+/**
+ * @internal
+ */
+class Configuration
+{
+    const GROUPS = ['total', 'indirect', 'direct', 'self'];
+
+    /**
+     * @var int[]
+     */
+    private $thresholds;
+
+    /**
+     * @var string
+     */
+    private $regex;
+
+    /**
+     * @var bool
+     */
+    private $enabled = true;
+
+    /**
+     * @var bool
+     */
+    private $verboseOutput = true;
+
+    /**
+     * @param int[]  $thresholds    A hash associating groups to thresholds
+     * @param string $regex         Will be matched against messages, to decide
+     *                              whether to display a stack trace
+     * @param bool   $verboseOutput
+     */
+    private function __construct(array $thresholds = [], $regex = '', $verboseOutput = true)
+    {
+        foreach ($thresholds as $group => $threshold) {
+            if (!\in_array($group, self::GROUPS, true)) {
+                throw new \InvalidArgumentException(sprintf('Unrecognized threshold "%s", expected one of "%s"', $group, implode('", "', self::GROUPS)));
+            }
+            if (!is_numeric($threshold)) {
+                throw new \InvalidArgumentException(sprintf('Threshold for group "%s" has invalid value "%s"', $group, $threshold));
+            }
+            $this->thresholds[$group] = (int) $threshold;
+        }
+        if (isset($this->thresholds['direct'])) {
+            $this->thresholds += [
+                'self' => $this->thresholds['direct'],
+            ];
+        }
+        if (isset($this->thresholds['indirect'])) {
+            $this->thresholds += [
+                'direct' => $this->thresholds['indirect'],
+                'self' => $this->thresholds['indirect'],
+            ];
+        }
+        foreach (self::GROUPS as $group) {
+            if (!isset($this->thresholds[$group])) {
+                $this->thresholds[$group] = 999999;
+            }
+        }
+        $this->regex = $regex;
+        $this->verboseOutput = $verboseOutput;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isEnabled()
+    {
+        return $this->enabled;
+    }
+
+    /**
+     * @param mixed[] $deprecations
+     *
+     * @return bool
+     */
+    public function tolerates(array $deprecations)
+    {
+        $deprecationCounts = array_filter($deprecations, function ($key) {
+            return false !== strpos($key, 'Count') && false === strpos($key, 'legacy');
+        }, ARRAY_FILTER_USE_KEY);
+
+        if (array_sum($deprecationCounts) > $this->thresholds['total']) {
+            return false;
+        }
+        foreach (['self', 'direct', 'indirect'] as $deprecationType) {
+            if ($deprecationCounts['remaining '.$deprecationType.'Count'] > $this->thresholds[$deprecationType]) {
```

Comment:
this is broken if there is no such threshold

Reply:
same as above